File: App/Resources/AlgorithmDetails/insertionsort/insertionsort.go
package main

import (
	"fmt"
)

func sort(arr []int) []int {
	n := len(arr)
	for i := 1; i < n; i++ {
		j := i
		for j > 0 && arr[j-1] > arr[j] {
			arr[j], arr[j-1] = arr[j-1], arr[j]
			j--
		}
	}
	return arr
}

func main() {
	array := []int{0, 39, 21, 62, 91, 77, 14, 23,
		90, 69, 51, 81, 68, 83, 32, 56}
	fmt.Println(sort(array))
}

File: App/Resources/AlgorithmDetails/insertionsort/insertionsort.kt
fun sort(arr: Array<Int>) {
  val n = arr.size
  for (i in 1 until n) {
    var j = i
    while (j > 0 && arr[j - 1] > arr[j]) {
      val temp = arr[j]
      arr[j] = arr[j - 1]
      arr[j - 1] = temp
      j--
    }
  }
}

fun main() {
  var array = arrayOf<Int>(
    0, 39, 21, 62, 91, 77, 14, 23,
    90, 69, 51, 81, 68, 83, 32, 56,
  )
  sort(array)
  println("[%s]".format(array.joinToString(", ")))
}